### PR TITLE
feat(transformer/typescript): report error for export = <value>

### PR DESCRIPTION
**Pull request by Dunqing**

### CODE REVIEW

**Comment by Dunqing** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3021?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#3021** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3021?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#3020** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3020?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#3018** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3018?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by Boshen** [comment]
### Merge activity

* **Apr 18, 6:52 AM EDT**: @Boshen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/3021).

File: crates/oxc_transformer/src/typescript/module.rs [hunk 1]
```diff
@@ -70,4 +73,14 @@ impl<'a> TypeScript<'a> {
 
         Declaration::VariableDeclaration(variable_declaration)
     }
+
+    pub fn transform_ts_export_assignment(
+        &mut self,
+        export_assignment: &mut TSExportAssignment<'a>,
+    ) {
+        println!("{:?}", self.ctx.source_type());
```
Comment (milesj) [inline crates/oxc_transformer/src/typescript/module.rs:81]: @Dunqing missed this

Comment (Dunqing) [reply crates/oxc_transformer/src/typescript/module.rs:81]: Oh, I forgot to delete this.

**Pull request merged**